src/ImpactEvaluator.sol

```diff
@@ -88,45 +87,50 @@ contract ImpactEvaluator is AccessControl {
         Round storage round = openRounds[roundIndex];
         require(round.exists, "Open round does not exist");
 
-        for (uint i = 0; i < addresses.length; i++) {
-            round.addresses.push(addresses[i]);
-            round.scores.push(scores[i]);
-        }
+        uint sumOfScores = validateScores(scores, round.totalScores);
+        reward(addresses, scores);
+        round.totalScores += sumOfScores;
 
-        if (allScoresSubmitted(round.scores)) {
-            reward(round.addresses, round.scores);
-            cleanUpRound(round, roundIndex);
+        if (round.totalScores == MAX_SCORE) {
+            delete openRounds[roundIndex];
         }
     }
 
-    function cleanUpRound(Round storage round, uint roundIndex) private {
-        round.end = 0;
-        delete round.addresses;
-        delete round.scores;
-        round.exists = false;
-        delete openRounds[roundIndex];
-    }
-
-    function allScoresSubmitted(
-        uint64[] memory scores
-    ) private pure returns (bool) {
-        uint totalScore = 0;
+    function validateScores(
+        uint64[] memory scores,
+        uint scoresAlreadySubmitted
+    ) private pure returns (uint) {
+        uint64 sum = 0;
         for (uint i = 0; i < scores.length; i++) {
-            totalScore += scores[i];
+            sum += scores[i];
         }
-        return totalScore >= MAX_SCORE;
+        require(sum <= MAX_SCORE, "Sum of scores too big");
+        require(
+            sum + scoresAlreadySubmitted <= MAX_SCORE,
+            "Sum of scores including historic too big"
+        );
+        return sum;
     }
 
     function reward(
         address payable[] memory addresses,
         uint64[] memory scores
     ) private {
```

Comment:
this function's flow was changed a bit, because we now first need to check how much we're expected to pay out for this call, before we can check whether we have enough balance for it